### PR TITLE
[BugFix] Use forward captured rowsets for query

### PR DESCRIPTION
**Pull request by ZiheLiu**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Related to #3689.

## Problem Summary(Required) ：

The captured rowsets will still be deleted from `_version_graph` and `_rs_version_map`, and it just guarantee that the stale rowsets cannot be deleted and closed.

`TabletReader::prepare` will try to capture rowsets again and throw error when rowsets are stale.
Therefore, pass the forward captured rowsets to `TabletReader`.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

- [x] I have checked the version labels which the pr will be auto backported to target branch

### CODE REVIEW

**Comment by ZiheLiu** [comment]
@Mergify backport branch-2.3 branch-2.4 branch-2.5

**Review by murphyatwork — DISMISSED** [review]

**Review by kangkaisen — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

File: be/src/storage/tablet_reader.cpp [hunk 1]
```diff
@@ -57,14 +65,17 @@ void TabletReader::close() {
...
-    auto st = _tablet->capture_consistent_rowsets(_version, &_rowsets);
-    if (!st.ok()) {
-        _rowsets.clear();
-        std::stringstream ss;
-        ss << "fail to init reader. tablet=" << _tablet->full_name() << "res=" << st;
-        LOG(WARNING) << ss.str();
-        return Status::InternalError(ss.str().c_str());
+    Status st = Status::OK();
+    if (_rowsets.empty()) {
+        std::shared_lock l(_tablet->get_header_lock());
```
Comment (satanson) [inline be/src/storage/tablet_reader.cpp:70]: !_rowsets.empty()  typo?  is it tested?

Comment (ZiheLiu) [reply be/src/storage/tablet_reader.cpp:70]: No, non-empty `_rowsets` indicates that it is captured before creating this TabletReader.
Therefore, only empty `_rowsets` need to capture rowsets in this if block.

Comment (ZiheLiu) [reply be/src/storage/tablet_reader.cpp:70]: I'd like to add some comments.

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

File: be/src/storage/tablet_reader.cpp (hunk 1 above)
Comment (satanson) [reply be/src/storage/tablet_reader.cpp:71]: so it is a second-time chance rowsets capturing?  add comment to explain why, it looks so weird.

**Review by kangkaisen — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.2

**Pull request merged**